internal/graph/check.go

```diff
@@ -479,21 +534,23 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 
 			var handlers []CheckHandlerFunc
 			for {
+				response.ResolutionMetadata.DatabaseReads++
```

Comment:
I would expect the counter to increase on the query execution, not the iterator. What was the rationale there?

I think we're conflating "dispatched" subproblems with database reads.

Reply:
Huh? I'm counting database reads. A call to `iterator.Next()` is one database read

Reply:
I just realized that `iterator.Next()` may not be one database read in Auth0 FGA 😢

Reply:
Yeah, that's my point, sorry if that wasn't clear. `iterator.Next()` is not always representative of a database query. In this case it's consuming another batch of rows on an existing query cursor. In DynamoDB it would be every time we fetch a new batch of results, etc..

So we just need to be carefully how we evaluate it.

Reply:
Talked with @elbuo8, agreed to count only the calls to Read, ReadUserTuple and ReadUsersetTuple

Reply:
Right, we shouldn't be counting the `iter.Next()`. That's what I said above ☝️